src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private List<? extends Machine> machines = new ArrayList<>();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            machines = new BulldozerProducer().get();
+        } else if (type.equals(Excavator.class)) {
+            machines = new ExcavatorProducer().get();
+        } else if (type.equals(Truck.class)) {
+            machines = new TruckProducer().get();
+        } else {
+            return new ArrayList<>();
+        }
```

Comment:
Probably we can omit this case, as if type is none of the above, then `machines` list will be empty anyway.